packages/chain-streams/src/leader.js

```diff
@@ -0,0 +1,65 @@
+// @ts-check
+import { E, Far } from '@endo/far';
+import { DEFAULT_RETRY_CALLBACK } from './defaults.js';
+import { shuffle } from './shuffle.js';
+import { makePollingWatcher } from './watcher.js';
+
+/**
+ * Create a chain leader that rotates through a list of endpoints.
+ *
+ * @param {string[]} endpoints
+ * @param {import('./types.js').ChainLeaderOptions} leaderOptions
+ */
+export const makeRoundRobinLeader = (endpoints, leaderOptions = {}) => {
+  const { retryCallback = DEFAULT_RETRY_CALLBACK } = leaderOptions;
+
+  // Shuffle the RPC addresses, so that we don't always hit the same one as all
+  // our peers.
+  shuffle(endpoints);
+
+  let lastRespondingEndpointIndex = 0;
+
+  /** @type {import('./types.js').ChainLeader} */
+  const leader = Far('round robin leader', {
```

Comment:
If you’re measuring latency and error rate, [The Power of Two Random Choices](https://www.eecs.harvard.edu/~michaelm/postscripts/handbook2001.pdf)

Reply:
I don't want to read a paper right now.  Can you be more specific as to what you're recommending, so that I can at least capture it as a TODO?